### PR TITLE
Doc: update PHPDoc for TreeBehavior moveUp and moveDown

### LINKED ISSUE #16091: TreeBehavior::moveUp does not return false for the root element
### Description

Moving the root element up does not return false as promised by the [documentation](https://api.cakephp.org/3.10/class-Cake.ORM.Behavior.TreeBehavior.html#moveUp()).

```
        $entity = $this->Areas->get($id);
        $moved = $this->Areas->moveUp($entity, 1);
```
$moved is an entity object.

The entity fields are the same before and after the call to moveUp:
```
    [id] => 1
    [name] => ...name...
    [parent_id] => null
    [lft] => 1
    [rght] => 190
```

Logs:

```
2021-11-01 10:26:53 Debug: duration=0 rows=0 BEGIN
2021-11-01 10:26:53 Debug: duration=0 rows=0 SELECT `Area`.`lft` AS `Area__lft`, `Area`.`rght` AS `Area__rght` FROM `areas` `Area` WHERE ((`parent_id`) IS NULL AND (`rght`) < 1) ORDER BY `lft` DESC LIMIT 1 OFFSET 0
2021-11-01 10:26:53 Debug: duration=0 rows=0 SELECT `Area`.`lft` AS `Area__lft`, `Area`.`rght` AS `Area__rght` FROM `areas` `Area` WHERE ((`parent_id`) IS NULL AND (`rght`) < '1') ORDER BY `lft` ASC LIMIT 1
2021-11-01 10:26:53 Debug: duration=0 rows=0 COMMIT
```

### CakePHP Version

3.10.0

### PHP Version

7.4

### PR DESCRIPTION
**Pull request by LordSimal**
Fixes https://github.com/cakephp/cakephp/issues/16091

### CODE REVIEW

File: src/ORM/Behavior/TreeBehavior.php [hunk 1]
```diff
@@ -601,12 +601,12 @@ protected function _removeFromTree(EntityInterface $node)
...
      *
      * If the node is the first child, or is a top level node with no previous node
-     * this method will return false
+     * this method will return the same node without any changes
      *
      * @param \Cake\Datasource\EntityInterface $node The node to move
      * @param int|true $number How many places to move the node, or true to move to first position
      * @throws \Cake\Datasource\Exception\RecordNotFoundException When node was not found
-     * @return \Cake\Datasource\EntityInterface|false $node The node after being moved or false on failure
+     * @return \Cake\Datasource\EntityInterface|false $node The node after being moved or false if given number is < 1
```
Comment (othercorey) [inline src/ORM/Behavior/TreeBehavior.php]:
```suggestion
     * @return \Cake\Datasource\EntityInterface|false $node The node after being moved or false if `$number` is < 1
```

Comment (LordSimal) [reply src/ORM/Behavior/TreeBehavior.php]: I adjusted the text for both appearances in `moveUp` and `moveDown`

**Comment by othercorey** [comment]
@LordSimal This is just a suggestion. If you want to keep the commit list clean in a PR, you can rebase the changes instead merging the parent branch. Then, you don't get commits like "Merge branch 'cakephp:4.x' into doc-treebehavior" in the history.

**Comment by LordSimal** [comment]
> @LordSimal This is just a suggestion. If you want to keep the commit list clean in a PR, you can rebase the changes instead merging the parent branch. Then, you don't get commits like "Merge branch 'cakephp:4.x' into doc-treebehavior" in the history.

Alright, I will definitely do that in the future 😄 thanks for the tipp

**Pull request merged**